### PR TITLE
Form.fill() doesn't clear text inputs before calling sendKeys()

### PR DESCRIPTION
**Pull request by emaks**
```
    @Test
    public void test() {
        Map<String, Object> data = new HashMap<>();
        data.put("login", "email_login");

        MainPage page = new MainPage(new FirefoxDriver());

        page.open();
        page.fillLogin(data);
        Assert.assertEquals(data.get("login"), page.getLogin());
        page.fillLogin(data);
        Assert.assertEquals(data.get("login"), page.getLogin());
    }

    public class MainPage {
        @FindBy(css = "form[role=form]")
        private Form loginForm;

        WebDriver driver;

        public MainPage(WebDriver driver) {
            this.driver = driver;
            PageFactory.initElements(new ElementDecorator(this.driver), this);
        }

        public void open() {
            driver.get("http://www.yandex.ua/");
        }

        public void fillLogin(Map<String, Object> data) {
            loginForm.fill(data);
        }

        public String getLogin() {
            return loginForm.getWrappedElement().findElement(By.cssSelector("[name=login]")).getAttribute("value");
        }
    }

### CODE REVIEW

**Comment by artkoshelev** [comment]
test this please

**Comment by artkoshelev** [comment]
that's good, please squash this commits into one and we'll merge it

**Comment by aik099** [comment]
Looks like a code duplication. We have similar issue with other element in past and it was solved there. I think that we should be using corresponding element methods, e.g. `TextElement` to set their value instead of fallback to `sendKeys` and similar methods.

**Comment by emaks** [comment]
> please squash this commits into one and we'll merge it

done

**Comment by artkoshelev** [comment]
@emaks wouldn't this change brake filling TextArea elements?

**Comment by emaks** [comment]
@artkoshelev, please, see new implementation.
and I have one question: Why there are no exceptions in this cases: [1st] (https://github.com/yandex-qatools/htmlelements/blob/master/htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Form.java#L61), [2nd] (https://github.com/yandex-qatools/htmlelements/blob/master/htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Form.java#L68) ?

**Comment by artkoshelev** [comment]
i think it's better to use StringUtils.defaultString which is null-safe version of String

**Comment by artkoshelev** [comment]
@emaks, and answering your question about exceptions, i don't know the reasons behind this implementation, but at least it works exactly as described in javadoc =)

File: htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/TextInput.java [hunk 1]
```diff
@@ -61,4 +63,13 @@ public String getText() {
         }
         return enteredText;
     }
+
+    public void fill(String value) {
+        // Returns sequence of backspaces and deletes that will clear element
```
Comment (artkoshelev) [inline htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/TextInput.java]: please change comment style to multiline here

File: htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Form.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public void fill(Map<String, Object> data) {
         for (String key : data.keySet()) {
             WebElement elementToFill = findElementByKey(key);
             if (elementToFill != null) {
-                fillElement(elementToFill, data.get(key));
+                fillElement(elementToFill, ObjectUtils.toString(data.get(key)));
```
Comment (artkoshelev) [inline htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Form.java]: let's add test to describe this behavior (clearing field when null is passed)

Comment (emaks) [inline htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Form.java]: done

File: htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/TextInput.java (hunk 1 above)
Comment (emaks) [inline htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/TextInput.java]: done

**Comment by artkoshelev** [comment]
test this please

**Comment by emaks** [comment]
@artkoshelev  is there any additional problems with pull request?

**Comment by artkoshelev** [comment]
@emaks i just need some time to made a final review

**Comment by artkoshelev** [comment]
landed, thanks!
